src/cgutils.cpp

```diff
@@ -2035,6 +2035,8 @@ static void emit_memcpy_llvm(jl_codectx_t &ctx, Value *dst, MDNode *tbaa_dst, Va
     // If the types are small and simple, use load and store directly.
     // Going through memcpy can cause LLVM (e.g. SROA) to create bitcasts between float and int
     // that interferes with other optimizations.
+#ifndef JL_LLVM_OPAQUE_POINTERS
```

Comment:
I'm really tempted to just outright delete this because the original motivation was a missed vectorization due to an intervening pointer->pointer bitcast from 6 years ago. I think with opaque pointers, this motivation is invalid, so this should be fine.

Reply:
Yep, fair enough. I just went for the minimal change here, but I agree, it's probably fine to delete.